core/src/main/scala/kafka/server/DelayedOperation.scala

```diff
@@ -292,34 +309,46 @@ class DelayedOperationPurgatory[T <: DelayedOperation](purgatoryName: String,
    */
   private class Watchers(val key: Any) {
 
-    private[this] val operations = new LinkedList[T]()
+    private[this] val operations = new ConcurrentLinkedQueue[T]()
+    private[this] val approximateWatched = new AtomicInteger(0)
```

Comment:
Hmm, could we benchmark the two options by having a thread call `DelayedOperationPurgatory.watched` periodically (whatever we think a monitoring system would do for such metrics) while running the Purgatory benchmark?